tests/61push/06_get_pusher.pl

```diff
@@ -50,7 +52,8 @@
          assert_eq( $pusher->{device_display_name}, $device_display_name, "device_display_name");
          assert_eq( $pusher->{pushkey}, $pushkey, "pushkey");
          assert_eq( $pusher->{lang}, $lang, "lang");
-         assert_eq( $pusher->{data}{testcustom}, $customdata, "custom data");
```

Comment:
I think we still want to ensure that custom fields are passed through.

Looking at the [push gateway spec](https://matrix.org/docs/spec/push_gateway/r0.1.1#post-matrix-push-v1-notify), `data` that is sent to the push gateway should be:

> A dictionary of additional pusher-specific data. For 'http' pushers, this is the data dictionary passed in at pusher creation minus the `url` key.

I think the changes should be:

1. Keep the custom field.
2. Ensure the custom field, `format` are both passed through.
3. Ensure the `url` is not passed through.

Reply:
I think this looks OK, but I'm unsure if `url` should be there or not, going to see what others think.

Reply:
this is checking the response from `GET /r0/pushers`, rather than the data sent to the push gateway. This  looks correct to me.